### PR TITLE
Adds new form input types to marshaller

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith 

draft of marshaller support for new form input types.

@drealeed Can you provide me with a ddl url where i can test this functionality?

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

File: src/form/Input.js [hunk 1]
```diff
@@ -103,7 +103,9 @@
         var optionHTML = "";
         if (optionsArr.length > 0) {
             optionsArr.forEach(function (opt) {
-                optionHTML += "<option value='" + opt + "'>" + opt + "</option>";
+                var val = (typeof(opt) === "object" && opt.constructor === Array ? opt[0] : opt);
```
Comment (mlzummo) [inline src/form/Input.js]: change to instance of Array ?

File: src/form/Form.js [hunk 2]
```diff
@@ -39,7 +39,7 @@
                     .name("select-test")
                     .label("Select Test")
                     .type("select")
-                    .selectOptions(["A","B","C"])
+                    .selectOptions([["AAA"],["BBB"],["CCC"]])
```
Comment (mlzummo) [inline src/form/Form.js]: this also can be done as this which is why we check if type is array below: [["AAA","text to display"],["BBB",text to display2"],["CCC",text to display3"]]

**Comment by mlzummo** [comment]
Need to add new widget to allow multiple checkboxs in a group and a radio button feature to existing input

**Comment by GordonSmith** [comment]
Closing until its ready to be pulled.

**Pull request closed without merging**